types/recharts/index.d.ts

```diff
@@ -714,6 +715,13 @@ export interface XPadding {
 	left: number;
 	right: number;
 }
+export interface XLabel {
+	value: string;
+	position?: string;
```

Comment:
Position is not explicit a string. Based on https://github.com/recharts/recharts/blob/master/src/component/Label.js it could be a list of possible positions and a x,y object.